File: backend/src/main/java/com/crm/entity/Task.java
package com.crm.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 任务 / 日报 / 待办。
 */
@Data
@TableName("tasks")
public class Task {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String title;
    private String type;
    private String content;
    private Long ownerId;
    private String relatedType;
    private Long relatedId;
    private LocalDateTime dueAt;
    private LocalDateTime remindAt;
    private Boolean done;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: backend/app/schemas/task.py
"""任务相关 Schema。"""
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TaskCreate(BaseModel):
    title: str
    type: str = "todo"
    content: str | None = None
    related_type: str | None = None
    related_id: int | None = None
    due_at: datetime | None = None


class TaskOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    type: str
    content: str | None = None
    owner_id: int | None = None
    related_type: str | None = None
    related_id: int | None = None
    due_at: datetime | None = None
    done: bool
    created_at: datetime
    updated_at: datetime
